sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -112,7 +112,12 @@ class SimpleFunctionRegistry extends FunctionRegistry {
         throw new AnalysisException(s"undefined function $name")
       }
     }
-    func(children)
+    try {
+      func(children)
+    } catch {
+      case _: ClassCastException =>
```

Comment:
Can we catch all of `ClassCastException`? I think that only this exception for ` scala.Function?` is related to wrong number of arguments.
This code may unexpectedly catch `ClassCastException` that occurred in a function body of given UDF.